sdk/servicebus/Microsoft.Azure.ServiceBus/src/Message.cs

```diff
@@ -295,18 +295,26 @@ public override string ToString()
         /// <summary>Clones a message, so that it is possible to send a clone of an already received
         /// message as a new message. The system properties of original message
         /// are not copied.</summary>
+        /// <remarks>The <see cref="UserProperties"/> is a new dictionary containing references to the original entries.</remarks>
         /// <returns>A cloned <see cref="Message" />.</returns>
         public Message Clone()
         {
             var clone = (Message)this.MemberwiseClone();
-            clone.SystemProperties = new SystemPropertiesCollection();
 
             if (this.Body != null)
             {
                 var clonedBody = new byte[this.Body.Length];
                 Array.Copy(this.Body, clonedBody, this.Body.Length);
                 clone.Body = clonedBody;
             }
+
+            clone.SystemProperties = new SystemPropertiesCollection();
+            clone.UserProperties = new Dictionary<string, object>(this.UserProperties.Count);
+            foreach (var userProperty in this.UserProperties)
+            {
+                clone.UserProperties.Add(userProperty);
+            }
```

Comment:
nit: If you're not cloning the members of the dictionary, why not use the constructor overload instead?

```csharp
clone.UserProperties = new Dictionary<string, object>(this.UserProperties);
```

Reply:
Haha.. Thanks. Didnt realize that overload existed.